src/code_1a1f0_len_5390.c

```diff
@@ -4,17 +4,49 @@ INCLUDE_ASM("code_1a1f0_len_5390", get_defeated);
 
 INCLUDE_ASM("code_1a1f0_len_5390", set_defeated);
 
-INCLUDE_API_ASM("code_1a1f0_len_5390", GetCamLookAtObjVector);
+ApiStatus GetCamLookAtObjVector(ScriptInstance* script, s32 isInitialCall) {
+    // Almost certainly a CURRENT_CAM macro
+    script->varTable[0] = (&gCameras[gCurrentCameraID])->lookAt_obj[0];
+    script->varTable[1] = (&gCameras[gCurrentCameraID])->lookAt_obj[1];
+    script->varTable[2] = (&gCameras[gCurrentCameraID])->lookAt_obj[2];
+
+    return ApiStatus_DONE2;
+}
 
 INCLUDE_API_ASM("code_1a1f0_len_5390", HasMerleeCasts);
 
 INCLUDE_API_ASM("code_1a1f0_len_5390", OnDefeatEnemy);
 
-INCLUDE_API_ASM("code_1a1f0_len_5390", OnFleeBattleDrops);
+ApiStatus OnFleeBattleDrops(ScriptInstance* script, s32 isInitialCall) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    PlayerData* playerData = &gPlayerData;
+
+    if (isInitialCall) {
+        script->functionTemp[0] = 0;
+        script->functionTemp[1] = 40;
+        script->functionTemp[2] = 0;
+    }
+
+    script->functionTemp[2]++;
+    if (script->functionTemp[2] >= 5) {
+        if (rand_int(100) < 50) {
+            if (playerData->coins != 0) {
+                playerData->coins--;
+                make_item_entity_delayed(ItemId_COIN, playerStatus->position.x, playerStatus->position.y + playerStatus->colliderHeight,
+                                         playerStatus->position.z, 3, 0, 0);
+            }
+        }
+        script->functionTemp[2] = 0;
+    }
+
+    script->functionTemp[1]--;
+    return script->functionTemp[1] == 0;
```

Comment:
can the above line be removed and this one changed to ``return --script->functionTemp[1] == 0;`` ?